### PR TITLE
modernize & bugfixing

### PR DESCRIPTION
**Pull request by Currywurst**
modernize,
do not initialize empty strings (there are already empty),
cast correctly (for example bool),
simplyfy return codes,
performance improcements( const reference strings ),
remove warnings ,
remove unused code

bugfix: check for type (null) in EPG

### CODE REVIEW

**Comment by Currywurst** [comment]
I had to read in first, but i hope this code helps you. 👍 
So, what's the next feature? :) Let me know. 
And yep, please make an codereview. I made it myself and testet all things. Think looks good. 

In ZatData there was another bug fixed with this issue. It's called check for type (null) in EPG . Grab the Type from Rapidjson and check it if it was a string. It can be null, but it can contains data for the next update thread.

**Comment by rbuehlma** [comment]
Thanks for your input. Its good to have someone looking into the code knowing c++ better than I do :)

Before I do the review, could you ensure that it also builds on windows (see the build failure).
And please do not include the .DS_Store file. You can add it to .gitignore

**Comment by Currywurst** [comment]
Done. I made 2 commits (4b755c7 - remove .ds_store) and ( 704259c .ds_store). 
So should i make a new pull request now? or can you grab it manually? 
Sorry, this was my first request here on Github.

Hm ;) Yep, i code c++ for years. As i seen this Addon i must understand the code first, and now i have to rewrite much things first.

**Comment by rbuehlma** [comment]
Ok, then i guess I was on the right track. I would prefer to prevent such an issue for every field. Therefore I created this patch.
https://github.com/rbuehlma/pvr.zattoo/commit/17d8f626deb43a50251bb2ae7c0da073d81b91da#diff-b8d956c375c70685659d947a7367cac5R991

If you agree that this should solve the issue as well, I will merge my patch first. You can then do a rebase of this PR and Squash the commits into one and merge that one as well.

File: src/client.cpp [hunk 1]
```diff
@@ -16,23 +17,23 @@ using namespace ADDON;
...
+CHelper_libXBMC_pvr *PVR = nullptr;
 
-std::string zatUsername = "";
-std::string zatPassword = "";
+std::string zatUsername;
+std::string zatPassword;
 bool zatFavoritesOnly = false;
 bool zatAlternativeEpgService = false;
-bool streamType = 0;
+bool streamType = false;
```
[the remark this answers is not in the document]
Comment (rbuehlma) [reply src/client.cpp]: Will be later changed to enum.

**Comment by Currywurst** [comment]
Hm... The patch looks very complicated for the main problem. It's better to compare types within an method. 
The crash was already fixed with 2 lines, but include yours. We can make an rework on the weekend.

**Comment by rbuehlma** [comment]
Yeah, I replaced all similar cases with the wrapper function to prevent this segmentation fault for all cases.

**Comment by Currywurst** [comment]
Now at Home, i merge Yours now.
Think i make a new pull request.


<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Currywurst** [comment]
Merged yours with mine and testet it on Amlogic, MacOS, Linux. 
So can you grab it from my forked repository and merge it into yours?

**Pull request merged**